packages/hub/node-tests/@core/compiler-test.ts

```diff
@@ -488,12 +488,19 @@ if (process.env.COMPILER) {
         let { compiled } = await cards.load(`${realm}person`);
         let source = getFileCache().getModule(compiled.schemaModule.global, 'browser');
         expect(source).to.containsSource(`
+          #allValuesPromise;
+
           get allFields() {
             return ["lastName", "aboutMe", "fullName"];
           }
+
           get allValues() {
-            return Promise.all(this.allFields.map(field => this[field]));
+            if (!this.#allValuesPromise) {
```

Comment:
I'm not sure if this is something we need to consider now, but what if we need to rerun this? Do we need to wire-up clearing of the promise when it's done so that can happen?

Reply:
there is an open question in the CardModel around how long we hang on to the schema class. right now we just make a new one each time we get a field. but when we get into caching and mutating we'll need to think more deeply about this.